services/blockstorage/adapter/filesystem/index.go

```diff
@@ -64,54 +71,73 @@ func (i *blockHeightIndex) fetchBlockOffset(height primitives.BlockHeight) (offs
 	return
 }
 
+// assumes timeStamp(inOrderBlock) <= timeStamp(topBlock)
```

Comment:
maybe add a comment that this is ignoring blocks that are not yet fully synced? the comment is an invariant i'm not sure it expresses what you mean...